geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/AcceptorImpl.java

```diff
@@ -1522,6 +1487,47 @@ public void run2() {
     }
   }
 
+  private CommunicationMode getCommunicationMode_notSelector(Socket socket) throws IOException {
+    socket.setSoTimeout(this.acceptTimeout);
+    this.socketCreator.configureServerSSLSocket(socket);
+    byte communicationModeByte = (byte) socket.getInputStream().read();
+    if (communicationModeByte == -1) {
+      throw new EOFException();
+    }
+    socket.setSoTimeout(0);
+    return CommunicationMode.fromModeNumber(communicationModeByte);
+  }
+
+  private CommunicationMode getCommunicationMode_selector(Socket socket) throws IOException {
```

Comment:
Underscores are generally not accepted in method names. Could we rename to `getCommunicationModeForSelectorSocket`